### PR TITLE
feat(operator): custom boms

### PR DESCRIPTION
**Pull request by lwander**

### CODE REVIEW

**Comment by lwander** [comment]
This started off as more of a guide, but wound up being more of a reference page... Maybe I should move it, and then if users run into trouble, provide a copy/paste example for the guides section?

File: guides/operator/custom-boms.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+In this case, the `${CONFIG_INPUT_ROOT}` is `${HALCONFIG_DIR}/.boms`. As a
+result, the BOM will be found under
+`${HALCONFIG_DIR}/.boms/boms/${VERSION}.yml`.
+
+> `${HALCONFIG_DIR}` is typically `~/.hal`
+
+At this point, the configuration files for each service will by default be read
+from GCS, unless you modify the BOM to indicate that they should be sourced
+localy. This is done by prefixing the subcompent version with `local:` as well.
```
Comment (duftler) [inline guides/operator/custom-boms.md]: s/localy/locally

File: guides/operator/custom-boms.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    version: 0.7.5
+  vault:
+    version: 0.7.0
+artifactSources:
+  debianRepository: https://dl.bintray.com/spinnaker-releases/debians
+  dockerRegistry: gcr.io/spinnaker-marketplace
+  gitPrefix: https://github.com/spinnaker
+```
+
+Along with a service's version, we record cannonical configuration for each
```
Comment (duftler) [inline guides/operator/custom-boms.md]: Here and below: s/cannonical/canonical

File: guides/operator/custom-boms.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+title:  "Deploy Custom Spinnaker Builds"
+sidebar:
+  nav: guides
+---
+
+## The Bill of Materials (BOM)
+
+Spinnaker is composed of many microservices, each of which have their own
+version. When a Spinnaker release is made, the microservices are tested
+together ensure they interoperate correctly, and their versions are recorded in
```
Comment (duftler) [inline guides/operator/custom-boms.md]: s/ensure/to ensure that

Comment (duftler) [reply guides/operator/custom-boms.md]: s/and their/and then their

File: guides/operator/custom-boms.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+`gs://${BUCKET}/boms/${VERSION}.yml`, and a configuration file `${PROFILE}` for
+subcomponent `${SUBCOMPONENT}` at version `${SUBCOMPONENT_VERSION}` can be
+found at `gs://${BUCKET}/${SUBCOMPONENT}/${SUBCOMPONENT_VERSION}/${PROFILE}`.
+
+#### Disabling GCS reads
+
+You can also completely disable reads from GCS by setting
+`spinnaker.config.input.gcs.enabled: false` in
+`/opt/spinanker/config/halyard-local.yml`. Be sure to restart the Halyard
+daemon for this configuration to take effect: `hal shutdown && hal`.
```
Comment (dorbin) [inline guides/operator/custom-boms.md:178]: What's the effect of disabling read from GCS? Is this meant to go hand-in-hand with the next section, configuration on your filesystem?

File: guides/operator/custom-boms.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
+---
+layout: single
+title:  "Deploy Custom Spinnaker Builds"
+sidebar:
+  nav: guides
+---
+
+## The Bill of Materials (BOM)
+
+Spinnaker is composed of many microservices, each of which have their own
```
Comment (dorbin) [inline guides/operator/custom-boms.md]: s/have their own/has its own/

File: guides/operator/custom-boms.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+version. When a Spinnaker release is made, the microservices are tested
```
Comment (dorbin) [inline guides/operator/custom-boms.md]: "When a Spinnaker release is made..."

"When Spinnaker is built," maybe?

**Review by dorbin — COMMENTED** [review]
@lwander , yeah, I see your point about reference v. guide. But I think it's as much conceptual as reference. Unfortunately, we don't yet have a place in Reference *or* Concepts for this material (unless maybe  /reference/architecture/life-of-a-bom/ ?). So my first reaction is to keep it here where you put it.

**Review by dorbin — APPROVED** [review]
LGTM, after changes

File: guides/operator/custom-boms.md (hunk 4 above)
Comment (lwander) [reply guides/operator/custom-boms.md:178]: No, you can have both reads from GCS as well as read from the filesystem. The below section describes that with "the configuration files will by default be read from GCS, unless..."

Comment (lwander) [reply guides/operator/custom-boms.md:178]: For a little more context, the ability to load from the filesystem has two purposes: 1. like you pointed out, to not require GCS, and 2. to make it easy to configure custom bills of materials (even if some shared resources live in the cloud).

**Pull request merged**